modules/unsupported/mbstyle/src/main/java/org/geotools/mbstyle/layer/SymbolMBLayer.java

```diff
@@ -1732,10 +1746,15 @@ public Expression textTranslateAnchor() {
                 "symbol-spacing", getId());
         symbolizer.getOptions().put("repeat", String.valueOf(symbolSpacing));
 
-        // text max angle
-        // layer.getTextMaxAngle();
-        // symbolizer.getOptions().put("maxAngleDelta", "40");
-
+        // text max angle - only for line placement
+        // throw MBFormatException if point placement
+        if(labelPlacement instanceof LinePlacement){
+            // will be true if line placement, false else.
+            symbolizer.getOptions().put("followLine", String.valueOf(followLine));
+            symbolizer.getOptions().put("maxAngleDelta", String.valueOf(getTextMaxAngle()));
+        }else if(labelPlacement instanceof PointPlacement && hasTextMaxAngle()){
+        	throw new MBFormatException("Text labels using point placement cannot utilize maxAngleDelta");
```

Comment:
Looking at the code, the format exceptions seem to typically be thrown by parse. However, since this depends on two properties, I think it is fine here.

Maybe update the wording a bit to better match the error messages in MBObjectParser and the parameters defined in the MapBox spec. How about:

```
"Property text-max-angle requires symbol-placement = line but symbol-placement = " + symbolPlacementVal
```

We'll probably need to revisit this when symbol placement supports functions.